src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs

```diff
@@ -0,0 +1,90 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration;
+using SonarQube.Client;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.Taint
+{
+    internal interface ITaintIssuesSynchronizer
+    {
+        /// <summary>
+        /// Fetches taint vulnerabilities from the server, converts them into visualizations and populates <see cref="ITaintStore"/>.
+        /// </summary>
+        Task Sync();
+    }
+
+    [Export(typeof(ITaintIssuesSynchronizer))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class TaintIssuesSynchronizer : ITaintIssuesSynchronizer
+    {
+        private readonly ITaintStore taintStore;
+        private readonly ISonarQubeService sonarQubeService;
+        private readonly ITaintIssueToIssueVisualizationConverter converter;
+        private readonly IConfigurationProvider configurationProvider;
+        private readonly ILogger logger;
+
+        [ImportingConstructor]
+        public TaintIssuesSynchronizer(ITaintStore taintStore,
+            ISonarQubeService sonarQubeService,
+            ITaintIssueToIssueVisualizationConverter converter,
+            IConfigurationProvider configurationProvider,
+            ILogger logger)
+        {
+            this.taintStore = taintStore;
+            this.sonarQubeService = sonarQubeService;
+            this.converter = converter;
+            this.configurationProvider = configurationProvider;
+            this.logger = logger;
+        }
+
+        public async Task Sync()
+        {
+            var bindingConfiguration = configurationProvider.GetConfiguration();
+
+            if (bindingConfiguration.Mode == SonarLintMode.Standalone)
+            {
+                logger.WriteLine(TaintResources.Synchronizer_NotInConnectedMode);
+                return;
+            }
+            try
+            {
+                var projectKey = bindingConfiguration.Project.ProjectKey;
+                var taintVulnerabilities = await sonarQubeService.GetTaintVulnerabilitiesAsync(projectKey, CancellationToken.None);
+
+                logger.WriteLine(TaintResources.Synchronizer_NumberOfServerIssues, taintVulnerabilities.Count);
+
+                var taintIssueVizs = taintVulnerabilities.Select(converter.Convert).ToList();
```

Comment:
`ToArray` rather than `ToList`?

It's an interesting question whether there's any value in realising the list in this case; one the one hand, we're only going to iterate through it once, so by create a list/array we're just creating an extra object that needs to be garbage collected.
One the other hand, knowing we're only going to iterate through it once is dependent on the current implementation of the store, which could change.
In practice it won't make much difference as this code isn't performance or memory critical.

Reply:
Well, it's more for the tests really. Otherwise I can't assert conversion failures 😛 
Probably won't make a difference.